### PR TITLE
HOCS-4268 Improve logging for user already exists

### PR DESCRIPTION
**Pull request by robertdeniszczyc2**

### CODE REVIEW

File: server/middleware/user.js [hunk 1]
```diff
@@ -56,13 +55,7 @@ async function addUser(req, res, next) {
...
         res.send({ userUUID: response.data.userUUID });
     } catch (error) {
-        if (isAxiosErrorWithCode(error, 409)) { // 409 errors are thrown when the user already exists and are logged in info service
-            logger.warn(error);
-            next(new UserAlreadyExistsError(error.response.data, error.response.status));
-        } else {
-            logger.error(error);
-            next(new FormSubmissionError(error.response.data, error.response.status));
-        }
+        next(new FormSubmissionError(error.response.data, error.response.status));
```
Comment (zuzak) [inline server/middleware/user.js:58]: Is this just going to display the error from info-service to the user? it looks like the info-service error sends the UUID, which isn't the best error message. Perhaps it should just silently succeed and redirect to the existing user's management page?

Comment (robertdeniszczyc2) [reply server/middleware/user.js:58]: info-service will throw back the message provided in the Exception, so in this case "User with provided email address already exists"

Line 66 on https://github.com/UKHomeOffice/hocs-info-service/pull/446/files#diff-55c46c9183cdfc580cfdc660eb2f524aeaa6cdeafd0e30573e362226cf1f8480

File: server/middleware/request.js [hunk 2]
```diff
@@ -10,10 +10,7 @@ function apiErrorMiddleware(err, req, res, _) {
     if (err instanceof ValidationError) {
         logger(req.requestId).info('VALIDATION_FAILED', { errors: Object.keys(err.fields) });
         return res.status(err.status).json({ errors: err.fields });
-    } else if (err instanceof UserAlreadyExistsError) {
-        logger(req.requestId).warn('USER_ALREADY_EXISTS', { errors: Object.keys(err.fields) });
-        res.status(409);
-    }  else {
+    } else {
         logger(req.requestId).error('ERROR', { message: err.message, stack: err.stack });
```
Comment (LiamMacP) [inline server/middleware/request.js:14]: Is this going to throw a log error out now?

Comment (robertdeniszczyc2) [reply server/middleware/request.js:14]: The error which actually throws and logs is in addUser() in user.js:

```
2022-02-18T13:10:33.407Z - error:  event_id=ERROR, message=User with provided email address already exists, stack=FormSubmissionError: User with provided email address already exists
    at addUser (<rest of path>/server/middleware/user.js:58:14)

```

Comment (LiamMacP) [reply server/middleware/request.js:14]: I'm happy with that for now - unsure if it should actually log at ERROR though.

**Review by LiamMacP — DISMISSED** [review]
Approved - change looks good

**Review by nickmcmahon01 — APPROVED** [review]

**Pull request merged**